Tuning Passenger configuration file

Aethylred opened the pull request:
This patch modifies `apache::mod::passenger` so that it creates a `passenger.conf` in the modules directory using a template. These are global parameters for the passenger instance, hence should be attached to the module and not to a vhost configuration.

Only a subset of the parameters have been created, these are sufficient for setting up a puppetmaster with apache::mod::passenger.

Commit:
Add parameter for purging vdir

The new parameter $purge_vdir in apache class gives the possibility to disable the purging of the virtual hosts directory.

blkperl committed:
Enable puppet 3.0.1 in travis.yml

A fix for rodjek/rspec-puppet#58 has been merged which
should allow puppet 3.0.1 to run the spec tests

kbarber committed:
Build all branches on travis

We should build all branches for travis, otherwise users won't be able to test
their own feature branches.

Signed-off-by: Ken Barber <ken@bob.sh>

kbarber committed:
Merge branch 'update_travis'

* update_travis:
  Build all branches on travis
  Enable puppet 3.0.1 in travis.yml

blkperl committed:
Standardize travis and gemfile

This commit standardizes the travis and gemfile to be similar
to the files in stdlib.

kbarber committed:
Merge pull request #144 from blkperl/update_travis

Standardize travis and gemfile

Commit:
...adding attribution and license

Commit:
add a location to store mode configurations for Debian & Ubuntu operating systems

Commit:
apache::mod::passenger now has parameters with default values and uses a template to configure passenger.

Commit:
file now requires the apache::mod

Commit:
a2mod do not need purging in Debian based Linux

Commit:
Need to be able to set RackBaseURI in the vhost

richardc committed:
Restrict the versions and add 3.1

~> 2.6 will match 2.7

kbarber committed:
Merge pull request #159 from richardc/patch-1

Restrict the versions and add 3.1

Aethylred commented:
excellent, and I've correctly identified a common node to rebase onto dc11bfdfd32020a7ede693c280a69df601c476ee :)

Commit:
Removing hard tabs

hunner commented:
Thanks @Aethylred! I'll get to merging this as soon as we have the current branch stabilized.

Note for myself when I get to merging this: rebase onto released refactored branch and add parameter descriptions to Lauren's updated docs.

Aethylred commented on manifests/init.pp:
```diff
@@ -65,8 +65,10 @@
     } -> A2mod <| |>
-    resources { 'a2mod':
-      purge => true,
+    if $::osfamily != 'debian' {
```
This block may no longer be required.

Commit:
Fix typos in README.passenger.md

bodepd committed:
Merge pull request #139 from mjanser/patch-1

Add parameter for purging vdir

Commit:
Capitalize RedHat to be in-line with actual factor output

Commit:
Fix vhost defined type unit test context to reflect actual test being performed.

context lists 'Debian' but test is for RedHat.

adrienthebo committed:
Merge pull request #171 from wolfspyre/master

Small fix to unit tests

Commit:
release 0.6.0

bodepd committed:
Merge pull request #174 from bodepd/0_6_0

release 0.6.0

trlinkin committed:
Custom logging for apache::vhost

This patch adds the ability to change how vhost logging works. This patch
adds new parameters to 'apache::vhost' define type. This allows users to
specify a log format or nickname for the 'CustomLog' directive. This also
allows users to specify a pipe as the destination for the 'ErrorLog' and
'CustomLog' directives.

hunner committed:
Merge branch 'refactor_module' into master

Conflicts:
	.travis.yml
	manifests/init.pp
	spec/defines/mod_spec.rb
	spec/defines/vhost_spec.rb

trlinkin committed:
Spec tests for apache::vhost logging improvements

The enhancements to vhost logging are currently passing the previous
spec tests in place. However the enhancements added two new ways the
apache::vhost type should fail. The enhancements also added some extra
logic surrounding the CustomLog and ErrorLog directives.

This patch extends the suite of tests already running against
apache::vhost logging enhancements added in 4226690eb3. The new
tests verify the correct functionality of the logging enhancements.

trlinkin committed:
Correct regexp warning in vhost_spec.rb

Without this patch, running the spec tests encounter a warning
about a regular expresion with a '}' that has not been escaped. This
warning is from line 238 in the vhost_spec.rb file.

This commit eliminates that warning.

trlinkin committed:
Add documentation for apache::vhost logging

Commit 6f4a317 added new paramters for controlling logging for the
apache::vhost type. This commit adds documentation for those changes.

hunner committed:
Merge pull request #183 from trlinkin/master

Remove spec test warning

trlinkin committed:
Refactor apache::vhost log file destination logic

This commit combines the blocks of logic determining the apache::vhost
logging values.

hunner committed:
Merge pull request #184 from trlinkin/refactor_vhost_logformat

Refactor vhost logformat

trlinkin committed:
Add Validation and rspec tests for $error_log

Without this commit, there is no validation or testing for the
$error_log parameter in the apache::vhost type.

This commit add boolean validation to the apache::vhost type. It also
adds two rspec tests to test the proper function of the parameter.

Commit:
Merge branch 'master' into passenger_tuning

Conflicts:
	manifests/init.pp
	manifests/mod/passenger.pp
	manifests/params.pp
	manifests/vhost.pp
	templates/mod/passenger.conf.erb
	templates/vhost-default.conf.erb

Commit:
Update passenger parameters to match standards in existing Apache module

Commit:
Fix some underscore placement issues

hunner committed:
Merge pull request #186 from trlinkin/refactor_vhost_logformat

Add Validation and rspec tests for $error_log

Commit:
update test to check passenger.conf file, and that it gets the correct path.

Commit:
ah, - not _ in mods-available

Aethylred commented:
ok.... this looks a bit messy, I've merged it with the new master from puppetlabs/puppetlabs-apache 

...and added some spec tests.

If you update refactor_module it should clean up.

Commit:
trying to test some of the passenger parameters

Commit:
testing that the passenger.conf contains passenger and ruby root directories, and no other declarations.

...there is possibly a better way of doing this...

Commit:
simplified regex....

Commit:
Fix Red Hat ruby root. Remove tests for things that should be missing.

Commit:
starting testing parameters

Commit:
This should provide a complete set of parameter tests.

Commit:
only _one_ typo in that last lot O_o

Aethylred commented:
That should be the complete testing that whatever you pass to the parameter ends up in the passenger.conf file.

nanliu committed:
Fix mod::prefork dependencies

A dependency on the package resolves the following error:
"Error: /Stage[main]/Apache::Mod::Prefork/File_line[/etc/sysconfig/httpd
prefork enable]: Could not evaluate: No such file or directory -
/etc/sysconfig/httpd"

trlinkin committed:
Move mpm_worker 'ServerLimit' line in ERB

It turns out that the order of MPM directives is important in certain
cases. Without this commit, the 'ServerLimit' directive was being
processed too late to be used in calculations for 'ThreadsPerChild'.

This commit moves the 'ServerLimit' directive to the top of the
directives in the ERB. This was based on Apache documentation and an
example config of theirs.

hunner committed:
Merge pull request #191 from trlinkin/master

Move 'ServerLimit' line in worker.erb

tjikkun committed:
Add proxy_pass parameter

mxey committed:
Use correct template for dav_fs module

hunner committed:
Merge pull request #188 from nanliu/prefork

Fix mod::prefork dependencies

hunner committed:
Merge pull request #194 from mxey/master

Use correct template for dav_fs module

hunner committed:
Merge pull request #193 from oxilion/proxy_pass

Add proxy_pass parameter

pronix committed:
fix variable for latest puppet version

apenney committed:
Minor tweak to set this to false as it's a replaced parameter.

hunner committed:
Merge pull request #209 from apenney/vdirpurgefix

Minor tweak to set this to false as it's a replaced parameter.

hunner committed:
Merge pull request #208 from pronix/master

fix variables for latest puppet

Commit:
Merge branch 'master' into passenger_tuning
